src/app/image.cpp

```diff
@@ -4,58 +4,46 @@
 
 #include "app/image.hpp"
 #include <algorithm>
+#include <iostream>
+
+constexpr float STANDARD_MAX_COLOR = 0.05;
+constexpr float GAMMA = 2.2;
 
 Image::~Image() {
     if (texture_m != nullptr) {
         SDL_DestroyTexture(texture_m);
     }
+    delete[] m_pixels;
+
 }
 
 void Image::initialize(int width, int height, SDL_Renderer *renderer) {
-    r_channel_m.resize(width, std::vector<double>(height, 0.0));
-    g_channel_m.resize(width, std::vector<double>(height, 0.0));
-    b_channel_m.resize(width, std::vector<double>(height, 0.0));
-    a_channel_m.resize(width, std::vector<double>(height, 0.0));
-
     width_m = width;
     height_m = height;
+
+    m_pixels = new Uint32[width_m * height_m];
+    memset(m_pixels, 0, width_m * height_m * sizeof(Uint32));
+
     renderer_m = renderer;
 
     min_exposure_m = 1.0;
 
+    auto uint32_size = static_cast<int>(sizeof(Uint32));
+    m_pitch = width_m * uint32_size;
```

Comment:
No need to create it as separate variable
```suggestion
    m_pitch = width_m * static_cast<int>(sizeof(Uint32));
```

Reply:
Fixed